Integrations/integration-Endgame.yml

```diff
@@ -0,0 +1,783 @@
+                }
+            });
+        }
+        return task;
+    };
+
+    var resToEntry = function(results){
+        currentCommand = commandToResult[command];
+        var entries = [];
+        for (var j in currentCommand) {
+            var current = currentCommand[j];
+            var entry = {
+                Type: entryTypes.note,
+                Contents: results,
+                ContentsFormat: formats.json
+                ReadableContentsFormat : formats.markdown,
+                EntryContext : {}
+            };
+            entry.HumanReadable = tableToMarkdown(current.title, results, Object.keys(current.MDdata), undefined, headerTransform(current.MDdata));
+            entry.EntryContext[current.contextPath] = jsonToEntity(results, underscoreToCamelCase);
+            entries.push(entry);
+        }
+        return entries;
+    }
+
+    var results;
+    login();
+    switch (command) {
+        case 'test-module':
+            return 'ok';
+        case 'endgame-investigation-results':
+            results = getInvestigationResults(args.investigation_id);
+            logout();
+            return results;
+        case 'endgame-investigation-status':
+            return getInvestigationStatus(args.investigation_id)
+        case 'endgame-get-unmanaged-endpoints':
+            results = getUnmanagedEndpoints();
+            res = resToEntry(results.data);
+            res[0].EntryContext['EndGame(true)'] = {'TransactionID' : results.metadata.transaction_id};
+            logout();
+            return res;
+        case 'endgame-get-deployment-profiles':
+            results = getDeploymentProfiles();
+            break;
+        case 'endgame-get-endpoint-status':
+            results = getEndpointStatus(args);
+            break;
+        case 'endgame-create-sensor-profile':
+            results = createSensorProfile(args.name, args.transceiver, args.sensor_version, args.sensor_directory, args.api_key);
+            break;
+        case 'endgame-create-investigation':
+            results = createInvestigation(args.investigation_name, args.assign_to ? args.assign_to : params.username.identifier, args.sensors, args.endpoints, args);
+            break;
+        case 'endgame-get-investigations':
+            results = getInvestigations();
+            break;
+        case 'endgame-get-sensor':
+            results = getSensor(args.sensor_id);
+            break;
+        case 'endgame-deploy':
+            results = deploy(args.domain, args.username, args.password, args.api_key, args.endpoints_transaction_id, args.deployment_profile, makeArray(args.endpoint_ips), makeArray(args.endpoint_ids));
+            if(results.submitted){
+                results.deployment_profile = args.deployment_profile;
+                results.id = results.submitted;
+                results.domain = args.domain;
+                results.submitted = undefined;
+            }
+            break;
+    }
+    logout();
```

Comment:
what happen if the request fails? no logout?

Reply:
fixed